realm/realm-library/src/androidTest/java/io/realm/RealmConfigurationTests.java

```diff
@@ -796,6 +801,120 @@ public void rxFactory() {
         assertFalse(configuration2.getRxFactory() == dummyFactory);
     }
 
+    @Test
+    public void rxFactory_nullThrows() {
+        try {
+            configFactory.createConfigurationBuilder()
```

Comment:
Normally we would do this to prevent accidental bugs to trigger the wrong failure:

```
private RealmConfiguration.Builder builder =  configFactory.createConfigurationBuilder()
try {
  builder.rxFactory(null)
} catch (IllegalArgumentException e {
  //...
}
```